patches/tflite-micro.patch

```diff
@@ -0,0 +1,143 @@
+@@ -941,6 +956,8 @@ TfLiteStatus MicroAllocator::CommitStaticMemoryPlan(
+ 
+   size_t actual_available_arena_size =
+       memory_allocator_->GetAvailableMemory(kBufferAlignment);
++  // We have to account for the memory needed by the memory planner
++  memory_allocator_->CalculateMaxUsed(memory_planner_->GetMaximumMemorySize());
+ 
+   // Make sure we have enough arena size.
+   if (memory_planner_->GetMaximumMemorySize() > actual_available_arena_size) {
+diff --git tensorflow/lite/micro/simple_memory_allocator.cc tensorflow/lite/micro/simple_memory_allocator.cc
+index 08b6789..8c01b17 100644
+--- tensorflow/lite/micro/simple_memory_allocator.cc
++++ tensorflow/lite/micro/simple_memory_allocator.cc
+@@ -34,7 +34,8 @@ SimpleMemoryAllocator::SimpleMemoryAllocator(ErrorReporter* error_reporter,
+       buffer_tail_(buffer_tail),
+       head_(buffer_head),
+       tail_(buffer_tail),
+-      temp_(buffer_head_) {}
++      temp_(buffer_head_),
++      max_used_(0) {}
+ 
+ SimpleMemoryAllocator::SimpleMemoryAllocator(ErrorReporter* error_reporter,
+                                              uint8_t* buffer,
+@@ -82,6 +83,7 @@ TfLiteStatus SimpleMemoryAllocator::SetHeadBufferSize(size_t size,
+   head_ = aligned_result + size;
+   temp_ = head_;
+ 
++  CalculateMaxUsed();
+   return kTfLiteOk;
+ }
+ 
+@@ -99,10 +101,11 @@ uint8_t* SimpleMemoryAllocator::AllocateFromTail(size_t size,
+     return nullptr;
+   }
+   tail_ = aligned_result;
++  CalculateMaxUsed();
+   return aligned_result;
+ }
+ 
+-uint8_t* SimpleMemoryAllocator::AllocateTemp(size_t size, size_t alignment) {
++uint8_t* SimpleMemoryAllocator::AllocateTemp(size_t size, size_t alignment, bool affects_max) {
+   uint8_t* const aligned_result = AlignPointerUp(temp_, alignment);
+   const size_t available_memory = tail_ - aligned_result;
+   if (available_memory < size) {
+@@ -112,7 +115,11 @@ uint8_t* SimpleMemoryAllocator::AllocateTemp(size_t size, size_t alignment) {
+                          size, available_memory, size - available_memory);
+     return nullptr;
+   }
++
+   temp_ = aligned_result + size;
++  if(affects_max) {
++    CalculateMaxUsed();
++  }
+   return aligned_result;
+ }
+ 
+@@ -135,7 +142,7 @@ size_t SimpleMemoryAllocator::GetAvailableMemory(size_t alignment) const {
+ }
+ 
+ size_t SimpleMemoryAllocator::GetUsedBytes() const {
+-  return GetBufferSize() - (tail_ - temp_);
++  return max_used_;
+ }
+ 
+ size_t SimpleMemoryAllocator::GetBufferSize() const {
+@@ -146,4 +153,13 @@ uint8_t* SimpleMemoryAllocator::head() const { return head_; }
+ 
+ uint8_t* SimpleMemoryAllocator::tail() const { return tail_; }
+ 
++void SimpleMemoryAllocator::CalculateMaxUsed(size_t extraToBeAdded) {
++  size_t tail_used = buffer_tail_ - tail_;
++  size_t head_used = std::max(head_ - buffer_head_, temp_ - buffer_head_);
++  size_t m = tail_used + head_used + extraToBeAdded;
++  if (max_used_ < m){
++    max_used_ = m;
```

Comment:
I know this is fine but std::max seems sensible

Reply:
Yes, you're right. Will make the change.